components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -449,4 +516,143 @@ private Object getBlobObject(InputStream is) throws IdentityApplicationManagemen
         return null;
     }
 
+    public List<String> getSessionIdsFromDb(String type) {
+
+        List<String> cacheKeyList = null;
+
+        if (!enablePersist) {
+            return cacheKeyList;
+        }
+
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+        try {
+            cacheKeyList = new ArrayList<String>();;
+            connection = jdbcPersistenceManager.getDBConnection();
+            connection.setAutoCommit(false);
+            preparedStatement = connection.prepareStatement(SQL_SELECT_LOGGED_IN_SESSION_IDS);
+            preparedStatement.setString(1, type);
+            resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cacheKeyList.add(resultSet.getString(1));
+            }
+            if (preparedStatement != null) {
+                preparedStatement.close();
```

Comment:
Why close the preparedStatement here ? Because from finally block it is closed anyway.